### PR TITLE
ls: long format author, group and owner

### PR DESCRIPTION
**Pull request by tertsdiepraam**
This PR adds the options to customize what information is shown in long format regarding author, group & owner. Specifically it adds:
- `--author`: shows the author, which is always the same as the owner. GNU has this feature because GNU/Hurd supports a difference between author and owner, but I don't think Rust supports GNU/Hurd, so I just used the owner.
- `-G` & `--no-group`: hide the group information.
- `-o`: hide the group and use long format (equivalent to `-lG`).
- `-g`: hide the owner and use long format.

The `-o` and `-g` options have some interesting behaviour that I had to account for. Some examples:
- `-og` hides both group and owner.
- `-ol` still hides the group. Same behaviour with variations such as `-o --format=long`, `-gl`, `-g --format=long` and `-ogl`.
- They even retain some information when overridden by another format: `-oCl` (or `-o --format=vertical --format=long`) still hides the group.

My previous solution for handling the behaviour where `-l1` shows the long format did not fit with these additions, so I had to rewrite that as well.

The tests only cover the how many names (author, group and owner) are present in the output, so it can't distinguish between, for example, author & group and group & owner.

### CODE REVIEW

**Review by sylvestre — COMMENTED** [review]
please rustfmt your changes!

File: src/uu/ls/src/ls.rs [hunk 1]
```diff
@@ -699,32 +838,45 @@ fn display_item_long(
         Ok(md) => md,
     };
 
-    println!(
-        "{}{}{} {} {} {} {} {} {}",
-        get_inode(&md, config),
+    #[cfg(unix)]
+    {
+        if config.inode {
+            print!("{} ", get_inode(&md));
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:844]: Do you think you could add a test covering this?

File: src/uu/ls/src/ls.rs [hunk 2]
```diff
@@ -699,32 +838,45 @@ fn display_item_long(
...
-    } else {
-        "".to_string()
-    }
-}
-
-#[cfg(not(unix))]
-fn get_inode(_metadata: &Metadata, _config: &Config) -> String {
-    "".to_string()
+fn get_inode(metadata: &Metadata) -> String {
+    format!("{:8}", metadata.ino())
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:879]: here too?

File: src/uu/ls/src/ls.rs [hunk 3]
```diff
@@ -922,8 +1070,8 @@ fn display_file_name(
     config: &Config,
 ) -> Cell {
     let mut name = get_file_name(path, strip);
-    if config.format != Format::Long {
-        name = get_inode(metadata, config) + &name;
+    if config.format != Format::Long && config.inode {
+        name = get_inode(metadata) + &name;
```
Comment (sylvestre) [inline src/uu/ls/src/ls.rs:1074]: same?

**Review by sylvestre — COMMENTED** [review]
Impressive work :)

**Comment by tertsdiepraam** [comment]
I've added tests for the `inode` in long and short formats, which should cover all the cases you mentioned. It was a very good suggestion, because I actually missed a space in the short format :)

**Comment by sylvestre** [comment]
@tertsdiepraam 
```

---- test_ls::test_ls_order_time stdout ----
current_directory_resolved: 
touch: /tmp/.tmpsrEiFD/test-1
open(append): /tmp/.tmpsrEiFD/test-1
touch: /tmp/.tmpsrEiFD/test-2
open(append): /tmp/.tmpsrEiFD/test-2
touch: /tmp/.tmpsrEiFD/test-3
open(append): /tmp/.tmpsrEiFD/test-3
touch: /tmp/.tmpsrEiFD/test-4
open(append): /tmp/.tmpsrEiFD/test-4
open: /tmp/.tmpsrEiFD/test-3
run: /home/sylvestre/dev/debian/coreutils/target/debug/coreutils ls -al
stderr = ""
stdout = "drwxr-xr-x  2 sylvestre sylvestre   4096 2021-03-21 16:58 .\ndrwxrwxrwt 93 root root 131072 2021-03-21 16:58 ..\n-rw-r--r--  1 sylvestre sylvestre      1 2021-03-21 16:58 test-1\n-rw-r--r--  1 sylvestre sylvestre      2 2021-03-21 16:58 test-2\n-rw-r--r--  1 sylvestre sylvestre      3 2021-03-21 16:58 test-3\n-rw-r--r--  1 sylvestre sylvestre      4 2021-03-21 16:58 test-4\n"
run: /home/sylvestre/dev/debian/coreutils/target/debug/coreutils ls -t
stderr = ""
stdout = "test-4\ntest-3\ntest-2\ntest-1\n"
run: /home/sylvestre/dev/debian/coreutils/target/debug/coreutils ls -tr
stderr = ""
stdout = "test-1\ntest-2\ntest-3\ntest-4\n"
run: /home/sylvestre/dev/debian/coreutils/target/debug/coreutils ls -tu
stderr = ""
stdout = "test-4\ntest-3\ntest-2\ntest-1\n"
thread 'test_ls::test_ls_order_time' panicked at 'assertion failed: `(left == right)`
  left: `"test-4\ntest-3\ntest-2\ntest-1\n"`,
 right: `"test-3\ntest-4\ntest-2\ntest-1\n"`', /home/sylvestre/dev/debian/coreutils/tests/by-util/test_ls.rs:396:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

```
I am experiencing this issue locally. did you see that?

**Comment by tertsdiepraam** [comment]
@sylvestre I saw that once on on the Windows CI, so I figured that it would only happen there and I changed the test to reflect that. The only explanation I can think of on Debian is that it has something to do with the resolution for access time. So maybe increasing the durations at the start of the test would fix it? They were originally all 500 milliseconds.

**Comment by tertsdiepraam** [comment]
There's also this note:
> Note that not all platforms will keep this field update in a file’s metadata, for example Windows has an option to disable updating this time when files are accessed and Linux similarly has `noatime`.

from: https://doc.rust-lang.org/nightly/std/fs/struct.Metadata.html#method.accessed
Maybe that's where the issue comes from? If so, then I could add an additional `if` that checks whether the access time is actually updated and then does the appropriate test.

**Pull request merged**